hudi-client/src/test/java/org/apache/hudi/client/TestMultiFS.java

```diff
@@ -63,9 +63,7 @@ public void setUp() throws Exception {
 
   @AfterEach
   public void tearDown() throws Exception {
-    cleanupSparkContexts();
-    cleanupDFS();
-    cleanupTestDataGenerator();
+    cleanupResources();
```

Comment:
@garyli1019 : Is this one of the places where we are missing the cleanup of resources ?

Reply:
We didn't miss one yet but it's easy to miss, so I put all the clean up in one place.
The leak I found was `HoodieLogFileReader` and `HoodieWrapperFileSystem`. These instances were created from different classes and difficult to clean up. 
By setting `Xmx2g` is consuming less memory than `Xmx4g`, probably because of the more aggressive GC.

Reply:
@garyli1019 : Can you open a jira with information about the leak that you found in HoodieLogFileReader and HoodieWrapperFileSystem. We need to get to the top of this asap.